src/inmanta/data/__init__.py

```diff
@@ -2335,6 +2335,7 @@ def validate_cron(value: str, allow_empty: bool = True) -> str:
 AUTOSTART_AGENT_DEPLOY_SPLAY_TIME = "autostart_agent_deploy_splay_time"
 AUTOSTART_AGENT_REPAIR_INTERVAL = "autostart_agent_repair_interval"
 AUTOSTART_AGENT_REPAIR_SPLAY_TIME = "autostart_agent_repair_splay_time"
+AUTOSTART_AGENT_RESTORE_STATE = "autostart_agent_restore_state"
```

Comment:
I would like some input on the name, especially the prefix. "autostart" isn't all that relevant anymore, because it's the only kind of agent at the moment. But I thought it best to stick with it, simply for consistency with the other option names.

Reply:
Because afaik we don't really expose the name / concept of "scheduler" anywhere at the moment. I see it as an internal change: it's a component of what the user knows as "the agent". But that's just how I see it, we can change that, but then I think we need to add a bunch of documentation as well.

Reply:
We expose it on iso8 dev: https://docs.inmanta.com/inmanta-service-orchestrator-dev/8/architecture.html

Reply:
I like scheduler fast start.

Reply:
I moved away from that name for two reasons:
1. I'm not sure it's always faster. Depends on how many versions we have to walk.
2. It feels a bit too vague to me. At least this gives the user an impression of what it does, and it's tied to the help string and the motivation for disabling it. I mean, when you consider a user facing the problem of "scheduler has inconsistent state", "restore-state" immediately hints at its relevance, while "fast-start" is a lot more far fetched.

Regarding AUTOSTART_AGENT vs SCHEDULER, if we do in fact document it, I think I agree that SCHEDULER is more appropriate. The only part that bothers me a bit is the inconsistency with the other options, though they do act on a slightly different level, so overall I think it's for the best.